### ISSUE TITLE
[0.48] Commits to cherry-pick into stable release

### ISSUE THREAD

**Issue by grabbou** [facebook/react-native#15318]

**Comment by iegik** [comment]
Broken commit: https://github.com/facebook/react-native/commit/712b1dd20ab0de73594696103b73c461c2c97076

```
node_modules/react-native/Libraries/NativeAnimation/RCTNativeAnimatedModule.h:14:9: 'React/RCTUIManagerObserverCoordinator.h' file not found
```

```
    "react": "^16.0.0-alpha.12",
    "react-native": "0.47.1",
```

**Comment by iegik** [comment]
```
node_modules/react-native/Libraries/Text/RCTTextFieldManager.m:14:9: 'React/RCTShadowView+Layout.h' file not found
```

**Comment by grabbou** [comment]
@iegik you sure about that one? In order to release a beta, I had to compile it. That looks like a compile-time issue that I didn't encounter. Have you managed to resolve it in the meantime?

**Comment by iegik** [comment]
Checkout also https://github.com/facebook/react-native/pull/15365#issuecomment-320955151

**Comment by iegik** [comment]
I'm definitely not sure about that, but then I try to build the application, i felt in a lot of errors. Tried to clean, install again and again. No luck.

**Comment by willdawsonme** [comment]
This https://github.com/facebook/react-native/commit/c9aeaf675dcf0292d148950d0fb020eeb13d3af1 would be great to get in if possible.

**Comment by eliperkins** [comment]
I'd like to nominate `603cc48ceba001827d10231d51b4031c7768eef8`, as described in this comment: https://github.com/facebook/react-native/commit/603cc48ceba001827d10231d51b4031c7768eef8#commitcomment-23539206

**Comment by ajostergaard** [comment]
While not exactly critical would be really nice to get HMR to work across all connected clients simultaneously - I believe https://github.com/facebook/react-native/commit/8b2975ad7b27ce34dcf56836685fd54ddd62086c sorts this out. Nice incremental improvement to everybody's workflow. :)

**Comment by grabbou** [comment]
@eliperkins 603cc48ceba001827d10231d51b4031c7768eef8 is already in the rc version. I have cherry-picked the rest and sent out a new build to Circle.

@ajostergaard I am going to wait with the HMR one for few more days to let it settle in the master branch.

**Comment by eliperkins** [comment]
@grabbou Ah! My mistake, I meant 603cc48ceba001827d10231d51b4031c7768eef8 into `0.47.0`, not `0.48.0`. 😬 

Due to the changes made in a8c45ac1c38866df222434b2d99d2a18b908c45f, content size on `TextInput` is currently broken, without cherry-picking 603cc48ceba001827d10231d51b4031c7768eef8. This seems like a pretty serious regression that deserves a patch, especially since we know how to fix it, and the fix is in the next monthly release as well.

Are patches still being made to `0.47.0`? I didn't see another issue similar to this one for `0.47.0`. I can open another ticket and nominate that cherry-pick if you'd like to keep the chatter out of this one.

**Comment by ide** [comment]
Here's the current status of things:
- rc.1 contains:
  - cb1b1e58b372bea23a2302db2f4c0eba85b4a17a (Fixes unintended side effects caused by #14684 for SwipeableRow) 
  - c9aeaf675dcf0292d148950d0fb020eeb13d3af1 (Separate window dimensions change from orientation)
- HEAD (will go out with rc.2) contains:
  - ef23d2bdcff66897c5746d9ad5f043b2900d00fb (Show bundle download progress on iOS)

The HMR fix is sitting in master and either way it will go out with 0.49.0-rc.0.

The rest of the discussions looked unrelated to cherry picks for 0.48.

**Comment by grabbou** [comment]
Thanks @ide for the summary! @eliperkins in order to suggest a patch to 0.47, please post in #14840

**Comment by Emilios1995** [comment]
Hi, maintainers!
Would it be possible to include flow 0.53 in the next release?
(commit #90eaeb019b2056150cae541a59720daad21466ef)

**Comment by grabbou** [comment]
I am not really sure it's needed. Cherry-picks are meant to be used when we fix issues that got fixed in master in the meantime. Features and improvements can usually wait for the next release (`0.49`). For historical reasons, I am against importing anything related to `flow` as it touches many files across the project and has high potential of breaking.

Anyways, for now I am closing this as release is being promoted to stable.

**Comment by fungilation** [comment]
Where's the 0.48 changelog?

**Comment by grabbou** [comment]
Will be published shortly on releases tab.

**Comment by grabbou** [comment]
One of the commits by @janicduplessis was reverted as it caused iOS to crash. Fix has been submitted and is already in 0.49 - we need to cherry-pick that one later this week.

**Comment by ide** [comment]
Adding an Android scroll view fix (e964a7f4ef93fcadf5cfd1c269b77b13cb3be374) and iOS font fix (d3007b0fd268c0d69f92) to 0.48-stable and releasing as 0.48.2. The files that both commits touch haven't had any other changes to them since 0.48-stable. 

Also cherry-picking these into 0.49-stable so that we don't regress from 0.48 to 0.49 and am releasing an 0.49 RC (will cross-post in the 0.49 cherry-pick issue so we don't miss this).

**Comment by ide** [comment]
Cherry-picked two bridge-related fixes for reloading C++ bridge (https://github.com/facebook/react-native/commit/1f9fc0ee3a2964b067f0848e35603475e510067c#diff-a2a67635fffd7b690d14dc17ae563a71 and https://github.com/facebook/react-native/commit/7b770556ac6eb60094bdf59bfd14876b6e2a1307). CI is publishing 0.48.3.

**Comment by fungilation** [comment]
https://github.com/facebook/react-native/releases/tag/v0.48.3 is up but not on npm? At least not to `yarn outdated`

**Comment by grabbou** [comment]
0.48.3 build has finished, please double check. Chances are you might have been looking before it finished.

**Comment by fungilation** [comment]
Ya I guess it took an hour or so for npm to pick up the new release? But it came on `yarn`, thanks

**Comment by ide** [comment]
Cherry-picked a fix for Jest tests into 0.48.4:
- Add BlobModule native module mock to jest setup: https://github.com/facebook/react-native/commit/165f87673002dec640fe86486296cd0d0847b333

**Comment by grabbou** [comment]
I have scheduled the build once again. I believe it should land on npm this time.

**Comment by fungilation** [comment]
Nope, still nothing.

Shouldn't this sort of thing be scripted?

**Comment by ide** [comment]
It is scripted but the CI pipeline failed.

**Comment by fungilation** [comment]
And still failing. Not on npm

**Comment by fungilation** [comment]
Any update on when the releases train _for bug fixes_ will be fixed?

**Comment by grabbou** [comment]
We are aware of the bug and working on a fix. We managed to get last build
green, but the publish npm step failed. So getting closer. I will update
this issue and post on Twitter once it's up. Apologies for the delays
around it, but build issues are the worst (it involves many devs across
teams to fix the issue)

**Comment by fungilation** [comment]
Looks like it's fixed? 0.48.4 was on npm
